src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt

```diff
@@ -46,22 +46,18 @@ class CourtService(
   fun getCourtIds() = courtIds.split(",").toSet()
 
   @Transactional(readOnly = true)
-  fun getVideoLinkAppointments(appointmentIds: Set<Long>): Set<VideoLinkAppointmentDto> {
-    return videoLinkAppointmentRepository
-      .findVideoLinkAppointmentByAppointmentIdIn(appointmentIds)
-      .asSequence()
+  fun getVideoLinkAppointments(appointmentIds: Set<Long>): Set<VideoLinkAppointmentDto> =
+    videoLinkBookingRepository.findByAppointmentIds(appointmentIds)
       .map {
-        VideoLinkAppointmentDto(
-          id = it.id!!,
-          bookingId = it.bookingId,
-          appointmentId = it.appointmentId,
-          hearingType = it.hearingType,
-          court = it.court,
-          createdByUsername = it.createdByUsername,
-          madeByTheCourt = it.madeByTheCourt
-        )
-      }.toSet()
-  }
+        with(it) {
```

Comment:
finding this a bit difficult to read - it reads like only a single appointment can be retrieved from a booking?

Reply:
how is only one appointment returned for a booking?

Reply:
This is really a legacy method used by PSH. Previously it searched for VideoLinkAppointments using VideoLinkAppointmentRepository. Now court, madeByTheCourt have been moved to the parent VideoLinkBooking. The repository query returns any booking where any appointmentid matches supplied appointmentIds set.  Then have to work out which one(s) matched and return DTOs for each.

On reflection I think I could have done some sort of join from appointment to booking. to return an appointment with its booking, or , event better, constructed the DTO in the query. (I think I might try that).

Reply:
Reviewing this code myself I realise it has a bug.
Originally this method used VideoLinkAppointmentRepository to retrieve an appointment with it's bookingId, court, madeByCourt.  
Those properties have been moved to the parent VideoLinkBooking so the original query is no longer sufficient.  Each appointment must be retrieved with its parent booking (or at least the properties of the parent booking). Since there are three separate releationships between a booking and an appointment that is a bit ugly. 
If we were free to redesign the tables we might possibly have a composite PK for appointment of videoLinkBookingId + HearingType (perhaps). But we have to retain the surrogate key of VideoLinkAppointment because it is used by PSH.
The bug arises because the same VLB can be returned for up to 3 different VLA - where the relationship(s) are some combination of pre, main and post.  DTOs should be constructed and returned for all matching relationships, but as it stands the code will return at most one.